Userland/Libraries/LibJS/Runtime/VM.cpp

```diff
@@ -690,8 +699,14 @@ ThrowCompletionOr<Value> VM::call_internal(FunctionObject& function, Value this_
     if (auto* interpreter = interpreter_if_exists())
         callee_context.current_node = interpreter->current_node();
 
-    callee_context.this_value = function.bound_this().value_or(this_value);
-    append_bound_and_passed_arguments(callee_context.arguments, function.bound_arguments(), move(arguments));
+    if (is<BoundFunction>(function)) {
+        auto& bound_function = static_cast<BoundFunction&>(function);
+        callee_context.this_value = bound_function.bound_this();
+        append_bound_and_passed_arguments(callee_context.arguments, bound_function.bound_arguments(), move(arguments));
+    } else {
+        callee_context.this_value = this_value;
+        append_bound_and_passed_arguments(callee_context.arguments, {}, move(arguments));
+    }
```

Comment:
Can a bound function even appear here? Since we have that branch `if (is<BoundFunction>(function))` above which unconditionally returns.

Reply:
Good catch, thanks!